packages/grid/_modules_/grid/hooks/root/useGridApiMethod.ts

```diff
@@ -12,16 +12,25 @@ export function useGridApiMethod<T extends Partial<GridApi>>(
   const apiMethodsRef = React.useRef(apiMethods);
   const [apiMethodsNames] = React.useState(Object.keys(apiMethods));
 
-  React.useEffect(() => {
-    apiMethodsRef.current = apiMethods;
-  }, [apiMethods]);
-
-  React.useEffect(() => {
+  const installMethods = React.useCallback(()=> {
+    if(!apiRef.current) {
+      return;
+    }
     apiMethodsNames.forEach((methodName) => {
       if (!apiRef.current.hasOwnProperty(methodName)) {
         logger.debug(`Adding ${apiName}.${methodName} to apiRef`);
         apiRef.current[methodName] = (...args) => apiMethodsRef.current[methodName](...args);
       }
     });
-  }, [apiMethodsNames, apiName, apiRef, logger]);
+  }, [apiMethodsNames, apiName, apiRef, logger])
+
+  React.useEffect(() => {
+    apiMethodsRef.current = apiMethods;
+  }, [apiMethods]);
+
+  React.useEffect(() => {
+    installMethods();
+  }, [installMethods]);
+
+  installMethods();
```

Comment:
The issue with Preact comes from a missing method `(getLocaleText)` on `apiRef`, in this PR I have tried to install the methods at an earlier stage of the React component lifecycle. 
While that attempt didn't work, I believe that if we manage to build `apiRef` at an earlier stage of the init process then it would fix this first Preact issue.